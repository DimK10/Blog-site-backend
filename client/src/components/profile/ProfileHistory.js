import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';

const ProfileHistory = ({
    auth: {
        user: { history },
    },
}) => {
    return (
        // TODO - update auth data in redux
        <Fragment>
            <h1>Your History:</h1>
            <ul>
                {history.map((h) => {
                    let msg = '';

                    if (h.parent) {
                        msg = `You have ${h.action}d a ${h.type} in `;
                        return (
                            <li key={uuidv4()}>
                                <p>{msg}</p>
                                <Link to={`/post/${h.postId}`}>
                                    {' this post'}
                                </Link>
                            </li>
                        );
                    }

                    if (h.type === 'post') {
                        msg = `You have ${h.action}d a `;

                        return (
                            <li key={uuidv4()}>
                                <p>{msg}</p>
                                <Link to={`/post/${h.id}`}>new post</Link>
                            </li>
                        );
                    }

                    msg = `You have ${h.action}d a ${h.type}`;
                    return (
                        <li key={uuidv4()}>
                            <p>{msg}</p>
                        </li>
                    );
                })}
            </ul>
        </Fragment>
    );
};

ProfileHistory.propTypes = {
    auth: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
    auth: state.auth,
});

export default connect(mapStateToProps)(ProfileHistory);
