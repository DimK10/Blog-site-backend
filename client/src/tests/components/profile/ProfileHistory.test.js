import React from 'react';
import { shallow } from 'enzyme';
import ProfileHistory from '../../../components/profile/ProfileHistory';

let wrapped = shallow(<ProfileHistory />);

describe('ProfileHistory', () => {
    it('should render the ProfileHistory Component correctly', () => {
        expect(wrapped).toMatchSnapshot();
    });

    it('should show the text Profile History', () => {
        expect(wrapped.find('Fragment').text()).toEqual('Profile history');
    });
});
