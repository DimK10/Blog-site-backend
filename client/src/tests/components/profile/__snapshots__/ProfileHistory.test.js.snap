// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHistory should render the ProfileHistory Component correctly 1`] = `
<Fragment>
  Profile history
</Fragment>
`;
